File: pdfSample/app/src/main/java/com/dgyu/pdfSample/Util/Toast.java
package com.dgyu.pdfSample.Util;

import android.content.Context;

/**
 * created Yu 2020. 04. 10
 *
 */

public class Toast {
    private static final boolean SHOW_TOAST = true;

    private static android.widget.Toast toast;
    private static Toast instance;
    private static Context context;

    public static Toast getInstance(){
        if(instance == null)
            instance = new Toast();

        return instance;
    }

    public void init(Context context) {
        this.context = context;
    }

    public void toast(Object message) {
        toast(context, message);
    }

    public static void toast(Context context, Object message, boolean showDefault, boolean build, boolean isLong) {
        if (SHOW_TOAST || showDefault) {
            if (toast == null) {
                toast = new android.widget.Toast(context);
                toast.setDuration(android.widget.Toast.LENGTH_SHORT);
            }
            if (toast != null) {
                toast.cancel();
            }
            if (isLong) {
                toast.setDuration(android.widget.Toast.LENGTH_LONG);
            } else {
                toast.setDuration(android.widget.Toast.LENGTH_SHORT);
            }
            if (message != null && message.toString() != "") {
                if (build) {
                    toast.setText(buildMessage(message));
                } else {
                    toast.setText(message.toString());
                }
                toast.show();
            }
        }
    }

    public static void toast(Context context, Object message, boolean showDefault, boolean build) {
        if (build) {
            toast(context, buildMessage(message), showDefault, false, false);
        } else {
            toast(context, message, showDefault, false, false);
        }
    }

    public static void toast(Context context, Object message, boolean showDefault) {
        toast(context, message, showDefault, false, false);
    }

    public static void toast(Context context, Object message) {
        toast(context, message, false, false, false);
    }

    private static String buildMessage(Object message) {
        StackTraceElement stackTraceElement = Thread.currentThread().getStackTrace()[4];
        StringBuilder sb = new StringBuilder();
        sb.append("[");
        sb.append(stackTraceElement.getFileName());
        sb.append(" ");
        sb.append(stackTraceElement.getLineNumber());
        sb.append("] ");
        sb.append(message);

        return sb.toString();
    }
}

File: Nlp/app/src/main/java/com/dgyu/nlp/Util/Toast.kt
package com.dgyu.nlp.Util

import android.content.Context
import android.os.Build
import android.widget.Toast

/**
 * created Yu 2022. 10. 05
 *
 */

class Toast {
    private val SHOW_TOAST = true

    private var toast: Toast? = null
    private var toastDuration = 0

    fun toast(
        context: Context?,
        message: Any?,
        showDefault: Boolean,
        build: Boolean,
        isLong: Boolean
    ) {
        if (SHOW_TOAST || showDefault) {
            if (toast == null) {
                toast = Toast(context)
                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
                    toast!!.duration = Toast.LENGTH_SHORT
                } else {
                    toastDuration = Toast.LENGTH_SHORT
                }
            }
            if (toast != null) {
                toast!!.cancel()
            }
            if (isLong) {
                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
                    toast!!.duration = Toast.LENGTH_LONG
                } else {
                    toastDuration = Toast.LENGTH_LONG
                }
            } else {
                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
                    toast!!.duration = Toast.LENGTH_SHORT
                } else {
                    toastDuration = Toast.LENGTH_SHORT
                }
            }
            if (message != null && message.toString() !== "") {
                var messages: String? = ""
                messages = if (build) {
                    buildMessage(message)
                } else {
                    message.toString()
                }
                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.R) {
                    toast!!.setText(messages)
                    toast!!.show()
                } else {
                    Toast.makeText(context, messages, toastDuration).show()
                }
            }
        } else {
            Log().e(message.toString())
        }
    }

    fun toast(context: Context?, message: Any, showDefault: Boolean, build: Boolean) {
        if (build && SHOW_TOAST) {
            toast(context, buildMessage(message), showDefault, false, false)
        } else {
            toast(context, message, showDefault, false, false)
        }
    }

    fun toast(context: Context?, message: Any?, showDefault: Boolean) {
        toast(context, message, showDefault, false, false)
    }

    fun toast(context: Context?, message: Any?) {
        toast(context, message, false, false, false)
    }

    private fun buildMessage(message: Any): String? {
        val stackTraceElement = Thread.currentThread().stackTrace[4]
        val sb = StringBuilder()
        sb.append("[")
        sb.append(stackTraceElement.fileName)
        sb.append(" ")
        sb.append(stackTraceElement.lineNumber)
        sb.append("] ")
        sb.append(message)
        return sb.toString()
    }
}